local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+        Args:
+            -reference (LatLon): latitude and longitude of the next global waypoint
+            -sailbot_position (LatLon): lat and lon position of SailBot
+            -sailbot_speed (float): speed of the SailBot in km/h
+        """
+        self.reference = reference
+
+        # Position of the Sailbot in XY
+        self.sailbot_position = latlon_to_xy(reference, sailbot_position)
+
+        self.sailbot_speed = sailbot_speed
+
+        # This should be defined in child classes
+        self.collision_zone = Polygon(None, None)
+
+    def is_valid(self, reference: LatLon, point_latlon: LatLon) -> bool:
+        """
+        Checks if a point is contained the obstacle's interior
+
+        Args:
+            point (Point): a shapely Point object representing the point to be checked
+
+        Returns:
+            bool: True if the point is not within the obstacle's interior, False otherwise
+        """
+        point = latlon_to_xy(reference, point_latlon)
+
+        # contains() requires a shapely Point object as an argument
+        point = Point(point[0], point[1])
+        return not self.collision_zone.contains(point)
+
+
+class Boat(Obstacle):
+
+    """
+    This class describes boat objects which Sailbot must avoid.
+    Also referred to target ships or boat obstacles.
+
+
+    Attributes
+    ----------
+        -id (int): MMSI number of the boat
+        -speed_over_ground (float): speed of the boat, over ground, in knots
+        -collision_cone (Polygon): a shapely Polygon object representing the boat's collision box
+            -the collision_cone represents the boats hull via a region of possible positions
+            -It is shaped like a cone, with the tip at the boat's position and the base at the
+             furthest possible position of the boat in the next time step
+            -the collision_cone is orientated according to the course of the boat
+            -position (XY): x,y coordinates of the boat's center (converted from lat/lon)
+            -course_over_ground (float): course of the boat in degrees, clockwise from true north
+            (#TODO: check if we need to consider magnetic north)
+            -width (float): width of the boat in meters
+            -length (float): length of the boat in meters
+    Notes
+    ----------
+        The following information about a boat is obtained from AIS:
+            - Maritime Mobile Service Identity (9 digit int)
+            - latitude (float) latitude in degrees
+            - longitude (float) longitude in degrees
+            - speed over ground (float) speed in knots over ground
+            - course over ground (float) degrees measured clockwise from true north
+            - boat dimensions (float) width and length of the boat in meters
+            - rate of turn (float) ROT in AISROT scale -126 to +126 corresponding to
+                -708 to +708 deg/min
+
+            - other information is also available but not used here
```

Comment:
Scope: shouldn't be documented here. Docstrings should only include what is necessary to understand the implementation. Can you move this to `HelperAISShip.msg`